jooby-filewatcher/src/main/java/org/jooby/filewatcher/FileMonitor.java

```diff
@@ -0,0 +1,155 @@
+package org.jooby.filewatcher;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.OVERFLOW;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchEvent.Kind;
+import java.nio.file.WatchEvent.Modifier;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Injector;
+
+class FileMonitor implements Runnable {
+
+  /** The logging system. */
+  private final Logger log = LoggerFactory.getLogger(getClass());
```

Comment:
Class is a singleton, no need to use a static logger. Also, I read somewhere that static logger cause leaks within dynamic class loaders (like the one we use with jooby:run), so unless is strictly necessary I alway keep log as final instance field.